flake.nix

```diff
@@ -1,30 +1,123 @@
+    nixpkgs,
+    crane,
+    flake-utils,
+    rust-overlay,
+    advisory-db,
+    pre-commit-hooks,
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      craneLib = crane.mkLib nixpkgs.legacyPackages.${system};
+
+      pkgs = import nixpkgs {
+        inherit system;
+        overlays = [(import rust-overlay)];
+      };
+
+      filterMd = path: _type: null != builtins.match ".*md" path;
+      filterSrc = path: type: (filterMd path type) || (craneLib.filterCargoSources path type);
+
+      src = pkgs.lib.cleanSourceWith {
+        src = craneLib.path ./.; # The original, unfiltered source
+        filter = filterSrc;
+      };
+
+      buildInputs = with pkgs;
+        lib.optionals stdenv.isDarwin [
+          darwin.apple_sdk.frameworks.Security
+          pkg-config
+        ];
+
+      nativeBuildInputs = with pkgs;
+        lib.optionals stdenv.isDarwin [
+          # darwin-specific inputs
+          clang
+          libiconv
+        ];
+
+      # Build *just* the cargo dependencies, so we can reuse
+      # all of that work (e.g. via cachix) when running in CI
+      cargoArtifacts = craneLib.buildDepsOnly {inherit src buildInputs nativeBuildInputs;};
+
+      # Build ripsecrets itself, reusing the dependency artifacts from above.
+      mujmap = craneLib.buildPackage {
+        inherit cargoArtifacts src buildInputs nativeBuildInputs;
+        propagatedBuildInputs = [pkgs.notmuch];
+        doCheck = false;
+        meta = with pkgs.lib; {
+          description = "Bridge for synchronizing email and tags between JMAP and notmuch";
+          homepage = "https://github.com/elizagamedev/mujmap";
+          maintainers = [maintainers.lafrenierejm];
+          license = licenses.gpl3;
         };
-        rustPkgs = pkgs.rustBuilder.makePackageSet {
-          rustVersion = "1.61.0";
-          packageFun = import ./Cargo.nix;
+      };
+
+      pre-commit = pre-commit-hooks.lib."${system}".run;
+    in {
+      packages = {
+        inherit mujmap;
+        default = mujmap;
+      };
+
+      apps.default = flake-utils.lib.mkApp {drv = mujmap;};
+
+      # `nix flake check`
+      checks = {
+        audit = craneLib.cargoAudit {inherit src advisory-db;};
+
+        clippy = craneLib.cargoClippy {
+          inherit cargoArtifacts src nativeBuildInputs;
+          cargoClippyExtraArgs = "--all-targets -- --deny warnings";
+        };
+
+        doc = craneLib.cargoDoc {inherit cargoArtifacts src;};
+
+        fmt = craneLib.cargoFmt {inherit src;};
+
+        nextest = craneLib.cargoNextest {
+          inherit cargoArtifacts src nativeBuildInputs;
+          partitions = 1;
+          partitionType = "count";
         };
-      in
-      {
-        packages = rec {
-          mujmap = ((rustPkgs.workspace.mujmap {}).overrideAttrs(oa: {
-            propagatedBuildInputs = oa.propagatedBuildInputs ++ [ pkgs.notmuch ];
-          })).bin;
-          
-          default = mujmap;
+
+        pre-commit = pre-commit {
```

Comment:
maybe submit this in another PR ? I dont really like this pre-commit stuff. How can I disable it by default ? there is an upcoming official nix formatter, I would prefer using that to alejandra.

Reply:
> maybe submit this in another PR ?

Sure, I can remove the pre-commit hooks from this PR.

Reply:
Removed the pre-commit hooks in e30a8310221ada3b0a376c2fa7de5e9363646063.